tests/Main.hs

```diff
@@ -227,23 +238,53 @@ testGetServiceHealth = testCase "testGetServiceHealth" $ do
 testHealth :: TestTree
 testHealth = testGroup "Health Check Tests" [testGetServiceHealth]
 
+-- Name of existing health check we can rely on for tests that use the Session API
+serfHealth :: Text
+serfHealth = "serfHealth"
+
+-- list of names of service/node checks (which should exist in consul already)
+checkIds :: [Text]
+checkIds = [serfHealth]
+
+-- for requests to session API
+lockDelay :: Maybe a
+lockDelay = Nothing
+
 {- Session -}
 testCreateSession :: TestTree
 testCreateSession = testCase "testCreateSession" $ do
   client@ConsulClient{..} <- newClient
-  let req = SessionRequest Nothing (Just "testCreateSession") Nothing ["serfHealth"] (Just Release) (Just "30s")
+  let ttl = "30s"
+      req = SessionRequest lockDelay
+                           (Just "testCreateSession")
+                           localNode
+                           checkIds
+                           (Just Release)
+                           (Just ttl)
   result <- createSession client req Nothing
   case result of
     Just _ -> return ()
     Nothing -> assertFailure "testCreateSession: No session was created"
 
+
+oneSecond :: Int
+oneSecond = 1 * 1000
+
+--
 testGetSessionInfo :: TestTree
 testGetSessionInfo = testCase "testGetSessionInfo" $ do
   client@ConsulClient{..} <- newClient
-  let req = SessionRequest Nothing (Just "testGetSessionInfo") Nothing ["serfHealth"] (Just Release) (Just "30s")
+  let ttl = "30s"
+      req = SessionRequest lockDelay
+                           (Just "testGetSessionInfo")
+                           localNode
+                           checkIds
+                           (Just Release)
+                           (Just ttl)
   result <- createSession client req Nothing
   case result of
     Just x -> do
+      threadDelay oneSecond
```

Comment:
This one is wrong by 1000x, `threadDelay` takes microseconds (I think you spotted that further down in the use of `twelveSeconds).

Reply:
oops, so `oneSecond` needs another `* 1000`, OK

Reply:
using `sleep` now instead.